### DISCUSSION TITLE
Problem using host rules

### DISCUSSION THREAD

**Discussion by renewfresho** [renovatebot/renovate#30875] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

We have private gems hosted on github.com 
So bundler needs to authenticate.
We don't want to use https because we don't want the credentials to be stored into the lockfile. 
So we need to use the username and password - it's not possible to set environment variables that don't start with MEND_ so we would like to use the host rules. 
When we set up host rules like so:
```
  "hostRules": [
    {
      "matchHost": "github.com",
      "hostType": "rubygems",
      "username": "{{ secrets.MEND_GITHUB_USERNAME }}",
      "password": "{{ secrets.MEND_GITHUB_PASSWORD }}"
    }
  ],
```
We are still getting an auth problem:
```
Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists
```
Here's the repo and job you can inspect:

https://developer.mend.io/github/Fresho-Org/fresho-document-store/-/job/990c1a2c-e0f5-4bd3-90ac-990e50cdb2bd

Cheers,
Rene

### Logs (if relevant)

<details><summary>Logs</summary>

```

Host key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists

```

</details>

**Comment by rarkins** [comment]
Are the private gems in the same github org (Fresho-Org)? Are the repos which need to access it also private repos?

**Reply by JodyVanden** [reply]
The gem is hosted on github and is part of the Fresho-org
the repo that renovate access is also private (and also part of the fresho-org)

**Reply by JodyVanden** [reply]
I tried a few things:
- [exclude the gem](https://github.com/renovatebot/renovate/discussions/30609) but found out it wasn't going to work because of bundle
- tried to split in separate gemfiles (having one for public gems and one for private gems) but it feels too inconvenient
- tried to pass a token, but that requires fetching the gem with https. Which is also inconvenient at the time
- tried to pass down a username and password

**Reply by rarkins** [reply]
If both repos are private, and both have the app installed, then the token passed to Renovate will already have permissions to access both repos and you shouldn't need to have a host rules.

What's the problem you're trying to resolve? e.g. was it a lookup failure (Renovate) or a lock file generation error (bundler)?

**Reply by JodyVanden** [reply]
I haven't installed the renovate app on the gem's folder... maybe I should try this?

bundler is not able to clone the gem from the `Gemfile.lock`
here are the errors of the logs:
```
DEBUG: rawExec err (branch="renovate/patch")
{
  "err": {
    "cmd": "/bin/sh -c bundler lock --patch --strict --update pg pry-rails rack-cors rails rollbar rspec-rails rspec-wait sidekiq-scheduler terser",
    "stderr": "Retrying `git clone --bare --no-hardlinks --quiet --no-tags --depth 1 --single-branch -- ssh://**redacted**@github.com/Fresho-Org/fresho-monitoring.git /tmp/renovate/cache/others/bundler/cache/bundler/git/fresho-monitoring-af4d5e833ac17074528b1b57f315c0105eb37e80` due to error (2/4): Bundler::Source::Git::GitCommandError Git error: command `git clone --bare --no-hardlinks --quiet --no-tags --depth 1 --single-branch -- ssh://**redacted**@github.com/Fresho-Org/fresho-monitoring.git /tmp/renovate/cache/others/bundler/cache/bundler/git/fresho-monitoring-af4d5e833ac17074528b1b57f315c0105eb37e80` in directory /tmp/renovate/cache/others/bundler/cache/bundler/git/fresho-monitoring-af4d5e833ac17074528b1b57f315c0105eb37e80 has failed.\nHost key verification failed.\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\nRetrying `git clone --bare --no-hardlinks --quiet --no-tags --depth 1 --single-branch -- ssh://**redacted**@github.com/Fresho-Org/fresho-monitoring.git /tmp/renovate/cache/others/bundler/cache/bundler/git/fresho-monitoring-
    ```

**Reply by JodyVanden** [reply]
just tried to add the renovate app to the gem's repo and rerun renovate app in the first repo.
configuration is using username and password
Still no success with bundler

**Reply by rarkins** [reply]
What type of dependency is it? Is it a git dependency (e.g. https://github.com/Fresho-org/something) or on GitHub's package registry (rubygems.pkg.github.com)?

**Reply by JodyVanden** [reply]
the dependency is a git dependency

**Reply by JodyVanden** [reply]
Is it possible to store a ssh key to be able to access a github repo?
I think we have tried a lot of options and the only remaining options would be:
- pass a ssh key to access the gem
- split the gemfile in multiple gemfiles and gemfile.lock and have all private gems in one that renovate could ignore - seems too annoying for our ways of working
- move all our gems to be accessed through https with a token - seems also annoying in terms of way of working

**Reply by JodyVanden** [reply]
we got it working 🥳 
we changed the gemfile to only access through HTTPS and used the hostrules in the mend UI
```
git_source(:github) do |repo|
    "https://github.com/#{repo}.git"
end
```

Now we have to figure out how our devs are able to access the gem with a github token
![image](https://github.com/user-attachments/assets/e0c2c556-d2c3-45c5-ae21-e623023b4af2)

thank you for help!

**Reply by JodyVanden** [reply]
to overcome our switch from ssh to github token, we used the github CLI to easily retrieve our token and store it for bundler to use it
might be good to put that in the documentation for future devs 🚀 
```
gh auth login -h github.com --skip-ssh-key
export BUNDLE_GITHUB__COM=$(gh auth token)
```
